Lombiq.HelpfulLibraries.AspNetCore/Security/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,153 @@
+
+            await next();
+        });
+
+    /// <summary>
+    /// Adds a middleware that sets the <c>X-Content-Type-Options</c> header to <c>nosniff</c>.
+    /// </summary>
+    /// <remarks><para>
+    /// "The Anti-MIME-Sniffing header X-Content-Type-Options was not set to ’nosniff’. This allows older versions of
+    /// Internet Explorer and Chrome to perform MIME-sniffing on the response body, potentially causing the response
+    /// body to be interpreted  and displayed as a content type other than the declared content type. Current (early
+    /// 2014) and legacy versions  of Firefox will use the declared content type (if one is set), rather than performing
+    /// MIME-sniffing." As written in <a href="https://www.zaproxy.org/docs/alerts/10021/">the documentation</a>.
+    /// </para></remarks>
+    public static IApplicationBuilder UseNosniffContentTypeOptionsHeader(this IApplicationBuilder app) =>
+        app.Use(async (context, next) =>
+        {
+            const string key = "X-Content-Type-Options";
+
+            if (!context.Response.Headers.ContainsKey(key))
+            {
+                context.Response.Headers.Add(key, "nosniff");
+            }
+
+            await next();
+        });
+
+    /// <summary>
+    /// Adds a middleware that checks all <c>Set-Cookie</c> headers and replaces any with a version containing
+    /// <c>Secure</c> and <c>SameSite=Strict</c> modifiers if they were missing.
+    /// </summary>
+    public static IApplicationBuilder UseStrictAndSecureCookies(this IApplicationBuilder app)
+    {
+        static void UpdateIfMissing(ref string cookie, ref bool changed, string test, string append)
+        {
+            if (!cookie.ContainsOrdinalIgnoreCase(test))
+            {
+                cookie += append;
+                changed = true;
+            }
+        }
+
+        return app.Use((context, next) =>
+        {
+            const string setCookieHeader = "Set-Cookie";
+            context.Response.OnStarting(() =>
+            {
+                var setCookie = context.Response.Headers[setCookieHeader];
+                if (!setCookie.Any()) return Task.CompletedTask;
+
+                var newCookies = new List<string>(capacity: setCookie.Count);
+                var changed = false;
+
+                foreach (var cookie in setCookie.WhereNot(string.IsNullOrWhiteSpace))
+                {
+                    var newCookie = cookie;
+
+                    UpdateIfMissing(ref newCookie, ref changed, "SameSite", "; SameSite=Strict");
+                    UpdateIfMissing(ref newCookie, ref changed, "Secure", "; Secure");
+
+                    newCookies.Add(newCookie);
+                }
+
+                if (changed)
+                {
+                    context.Response.Headers[setCookieHeader] = new StringValues(newCookies.ToArray());
+                }
+
+                return Task.CompletedTask;
+            });
+
+            return next();
+        });
+    }
```

Comment:
> However, the method UpdateIfMissing could be improved by checking if the context is secure before appending the Secure attribute. This is because the Secure attribute should only be set if the request is served over HTTPS.

This middleware should not be used without HTTPS redirection enabled. I've added a remark in my last commit to explain that.

> Additionally, the SameSite attribute should be checked for a specific value rather than just the presence of the term "SameSite" to avoid overwriting an intentionally set Lax or None value.

`UpdateIfMissing(ref newCookie, ref changed, "SameSite", "; SameSite=Strict");`  means that it will only update the cookie if it contains no `SameSite` attribute at all. So the suggestion is not necessary.